### ISSUE TITLE
Coffeescript Syntax Highlighting: Shows code as comments

### ISSUE THREAD

**Issue by kellertobi91** [jneen/rouge#599]
### Summary

After a line with multiple hashtags, code is rendered as multiline comments.

```coffee
normalFunction = () ->
    console.log(stuff)

# Normal Comment
x = normalCode

###
Multiline Comment works
###

x = normalCode

###############################################
# Has more than three hashtags, so no multiline Comment  ##
###############################################

x = normalCode?

###############################################

x = normalCode?

###

```

### Steps to reproduce

See example above 

### Expected behavior

All `x=normalCode` lines rendered as code and not comments

### Actual behavior

After more than three hashtags code is rendered as comment.

**Comment by kellertobi91** [comment]
Using this in Gitlab: https://gitlab.com/gitlab-org/gitlab-ce/issues/26075#note_20496226

**Comment by dblessing** [comment]
@kellertobi91 What about the closing line? Is the following valid?

```coffeescript
### 
My comment

Random line of more than three hashes
############

###
```

**Comment by kellertobi91** [comment]
Well, github seems to render it correctly. Just try js2.coffee

```coffee
### 
My comment

Random line of more than three hashes
############

no_comment = ""

###
```

A multi line comment is started with excactly three hashes and a newline or space after and is finished with three or more hashes. This is how sublime renders it, this is how atom renders it, it is how js2.coffee renders it.

I just figured out that also the keyword `unless` is not recognized

**Comment by dblessing** [comment]
@kellertobi91 Thanks. I submitted a fix in #604. 

I checked that the lexer already accepts any number of hashes to close a multiline comment. It's actually a side-effect. The multiline rule matches the first three hashes to close the multiline and then the single comment rule catches the rest as a single line comment. Works for me!

**Comment by kellertobi91** [comment]
@dblessing thanks to _you!_ hope that it will be merged soon an gitlab updates to that version ;-)
